client/src/main/java/com/netflix/conductor/client/automator/PollingSemaphore.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.conductor.client.automator;
+
+import java.util.concurrent.Semaphore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class wrapping a semaphore which holds the number of permits available for polling and executing tasks.
+ */
+class PollingSemaphore {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PollingSemaphore.class);
+    private Semaphore semaphore;
+
+    PollingSemaphore(int numSlots) {
+        LOGGER.debug("Polling semaphore initialized with {} permits", numSlots);
+        semaphore = new Semaphore(numSlots);
+    }
+
+    /**
+     * Signals if polling is allowed based on whether a permit can be acquired.
+     *
+     * @return {@code true} - if permit is acquired
+     *         {@code false} - if permit could not be acquired
+     */
+    boolean canPoll() {
+        boolean acquired = semaphore.tryAcquire();
+        LOGGER.debug("Trying to acquire permit: {}", acquired);
```

Comment:
This log itself may not be helpful much. Should we use a metric instead?

Reply:
This is a debug level log which is intended to be used only for debugging issues, a metric here would not give any valuable info.